### ISSUE TITLE
Tile decals are rendered over other decals

### ISSUE THREAD

**Issue by Cheshify** [tgstation/tgstation#74165]
## Round ID:
N/A

## Testmerges:
Related to this PR but not necessarily caused by it
#74035

## Reproduction:
![image](https://user-images.githubusercontent.com/73589390/226776150-08aa8602-2990-46cf-ab3d-32ff80ac7008.png)

The layers defined in the code for tiles look like this ->
Tile decals use:     layer = TURF_PLATING_DECAL_LAYER
Other turfplating decals use: layer = TURF_DECAL_LAYER

This means that those square tiles should always be rendered under other decals, but that's not the case. What actually happens is that the order in which a mapper places down decals makes these tile decals render over others. According to lemon, this issue is because...

> those layers seem to only apply in editor. once the decal is applied, the overlay it creates just seems to... drop the old object's layer? like there's support for passing in layer here but it just isn't used, so they instead inherit it from the turf they're applied to 
> 
> that means that the layer value doesn't do anything, and instead what matters is the init order, or in what order the objects are placed on the map. highly stupid

Basically, the actual layer is ignored and init order is used instead. Mappers shouldn't need to order every single decal on every single tile, considering how many we use.

**Comment by san7890** [comment]
it has to do with how appearances are applied to the turf, since the turf_decal effect deletes itself after it colors the turf its resting on

i think i tried to do something to make this work using `/datum/element/decal/proc/generate_appearance()` (since that's what actually does the work) but for some reason it wasn't budging no matter what zany stuff I was doing with layers? maybe my coding power level was too low for me to understand what i was doing wrong

**Comment by LemonInTheDark** [comment]
san's out here digging at the belly of the beast and I just go haha funny arg go boop (if my pr doesn't fully fix this I am going to delete this comment and pretend it never happen)
